src/nvim/option.c

```diff
@@ -2596,6 +2596,8 @@ static char *did_set_string_option(int opt_idx, char_u **varp, bool new_value_al
         // only encoding=utf-8 allowed
         if (STRCMP(p_enc, "utf-8") != 0) {
           errmsg = e_unsupportedoption;
+        } else {
+          spell_reload();
```

Comment:
Perhaps we could call this from `:set all&` ? Then the test could do that, though it might require setting some other options or sourcing `testdir/setup.vim` again.

Then we could avoid this weird special case.